scrunch/mutable_dataset.py

```diff
@@ -0,0 +1,892 @@
+            element='shoji:entity',
+            body=dict(
+                alias=alias,
+                name=name,
+                derivation=expr,
+                description=description,
+                notes=notes
+            )
+        )
+
+        self.resource.variables.create(payload)
+        # needed to update the variables collection
+        self._reload_variables()
+        # return the variable instance
+        return self[alias]
+
+    def create_categorical(self, categories, alias, name, multiple,
+                           description='', notes=''):
+        """
+        Used to create new categorical variables using Crunchs's `case`
+        function
+
+        Will create either categorical variables or multiple response depending
+        on the `multiple` parameter.
+        """
+        if multiple:
+            return self.create_multiple_response(
+                categories, alias=alias, name=name, description=description,
+                notes=notes)
+        else:
+            return self.create_single_response(
+                categories, alias=alias, name=name, description=description,
+                notes=notes)
+
+    def create_crunchbox(
+            self, title='', header='', footer='', notes='',
+            filters=None, variables=None, force=False, min_base_size=None,
+            palette=None):
+        """
+        create a new boxdata entity for a CrunchBox.
+
+        NOTE: new boxdata is only created when there is a new combination of
+        where and filter data.
+
+        Args:
+            title       (str): Human friendly identifier
+            notes       (str): Other information relevent for this CrunchBox
+            header      (str): header information for the CrunchBox
+            footer      (str): footer information for the CrunchBox
+            filters    (list): list of filter names or `Filter` instances
+            where      (list): list of variable aliases or `Variable` instances
+                               If `None` all variables will be included.
+            min_base_size (int): min sample size to display values in graph
+            palette     dict : dict of colors as documented at docs.crunch.io
+                i.e.
+                {
+                    "brand": ["#111111", "#222222", "#333333"],
+                    "static_colors": ["#444444", "#555555", "#666666"],
+                    "base": ["#777777", "#888888", "#999999"],
+                    "category_lookup": {
+                        "category name": "#aaaaaa",
+                        "another category:": "bbbbbb"
+                }
+
+        Returns:
+            CrunchBox (instance)
+        """
+
+        if filters:
+            if not isinstance(filters, list):
+                raise TypeError('`filters` argument must be of type `list`')
+
+            # ensure we only have `Filter` instances
+            filters = [
+                f if isinstance(f, Filter) else self.filters[f]
```

Comment:
filters is a property that was requested, it's the only way for users to iterate over the filters of a dataset.